src/main/java/org/tikv/common/log/SlowLogSpanImpl.java

```diff
@@ -17,78 +17,79 @@
 
 package org.tikv.common.log;
 
-import static org.tikv.common.log.SlowLogImpl.getSimpleDateFormat;
-
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
 
 public class SlowLogSpanImpl implements SlowLogSpan {
   private final String name;
-  private final long requestStartNS;
-  private final long requestStartMS;
+  private final long requestStartInstantNS;
+  private final long requestStartUnixNS;
+
+  /** Key-Value pairs which will be logged, e.g. function name, key, region, etc. */
+  private final Map<String, String> properties;
 
-  private long startMS;
-  private long endMS;
   /**
    * use System.nanoTime() to calculate duration, cause System.currentTimeMillis() is not monotonic
    */
-  private long startNS;
+  private long startInstantNS;
 
-  private long endNS;
+  private long endInstantNS;
 
-  public SlowLogSpanImpl(String name, long requestStartMS, long requestStartNS) {
+  public SlowLogSpanImpl(String name, long requestStartMS, long requestStartInstantNS) {
     this.name = name;
-    this.requestStartMS = requestStartMS;
-    this.requestStartNS = requestStartNS;
-    this.startMS = 0;
-    this.startNS = 0;
-    this.endMS = 0;
-    this.endNS = 0;
+    this.requestStartUnixNS = requestStartMS * 1_000_000;
+    this.requestStartInstantNS = requestStartInstantNS;
+    this.properties = new HashMap<>();
+    this.startInstantNS = 0;
+    this.endInstantNS = 0;
+  }
+
+  @Override
+  public void addProperty(String key, String value) {
+    properties.put(key, value);
   }
 
   @Override
   public void start() {
-    startNS = System.nanoTime();
-    startMS = requestStartMS + (startNS - requestStartNS) / 1_000_000;
+    startInstantNS = System.nanoTime();
   }
 
   @Override
   public void end() {
-    endNS = System.nanoTime();
-    endMS = startMS + (endNS - startNS) / 1_000_000;
+    endInstantNS = System.nanoTime();
   }
 
   @Override
   public JsonElement toJsonElement() {
-    SimpleDateFormat simpleDateFormat = getSimpleDateFormat();
+    SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
     JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("name", name);
-    jsonObject.addProperty("start", getStartString(simpleDateFormat));
-    jsonObject.addProperty("end", getEndString(simpleDateFormat));
-    jsonObject.addProperty("duration", getDurationString());
+    jsonObject.addProperty("event", name);
+    jsonObject.addProperty("begin", dateFormat.format(getStartUnixNS() / 1_000_000));
+    jsonObject.addProperty("begin_unix_ms", getStartUnixNS() / 1_000_000);
+    jsonObject.addProperty("duration_ms", getDurationNS() / 1_000_000);
```

Comment:
would it result in a `long` value since both values are integers?

Reply:
begin_unix_ms is removed.